lib/temporal/testing.rb

```diff
@@ -23,6 +25,31 @@ def local?
         mode == LOCAL_MODE
       end
 
+      def execute_scheduled_workflow(workflow_id:)
+        unless scheduled_executions.key?(workflow_id)
+          raise Temporal::Testing::WorkflowIDNotScheduled,
+            "There is no workflow with id #{workflow_id} that was scheduled with Temporal.schedule_workflow.\n"\
+            "Options: #{scheduled_executions.keys}"
+        end
+
+        scheduled_executions[workflow_id].call
+      end
+
+      def execute_all_scheduled_workflows
+        scheduled_executions.transform_values(&:call)
+      end
+
+      # Populated by Temporal.schedule_workflow
+      # format: { <workflow_id>: <cron schedule string>, ... }
+      def schedules
```

Comment:
I might be good to have schedules live close to `executions` in the override module. One particular option I'm considering is being able to check if a workflow has been scheduled using Temporal's public API. I wonder how does that actually work today — will Temporal allow you to query a workflow that was scheduled, but not yet executed…

Reply:
Interesting.  It seems weird to put schedules and scheduled_executions  in different modules, but I could move them both back to the overrides.    (See my most recent stacked commit)
The reason I moved was for sorbet; I needed to add types to our `.rbi` files and I didn't want to add these types to the main `Temporal` namespace even though they aren't currently available in production.
Maybe we could move them over if/when the production version becomes available?  WDYT?

Reply:
Right, sorry, my bad. Definitely don't want to extend `Temporal`'s interface with testing methods, should be identical to the real production one. Makes sense to use `Temporal::Testing`. However it feels like the `Temporal::Testing` should call into some common private class to get the scheduled workflows and executions.

One of the smells here is essentially having publicly writable `Temporal::Testing.schedules` which can be updated from the test code directly.

Basically we need something to track schedules/executions that's neither `Temporal` nor `Temporal::Testing`, but used by both

Reply:
Agreed, will move into a private class.